### PR TITLE
Decouple deployment metrics from the metricservice

### PR DESCRIPTION
**Pull request by smorgrav**
This is to avoid external calls to the metric providers when asking for deployment info.

With this change the deployment info is self-contained and can be returned without any (potentially) external calls.

### CODE REVIEW

**Comment by smorgrav** [comment]
Will commit the maintainer before it is ready to be merged. @bratseth asking for your review as this was kind of a over-the-shoulder decision that you and Frode made If I understood it correctly.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java [hunk 1]
```diff
@@ -334,9 +334,10 @@ else if ( ! application.deploying().isPresent() && ! zone.environment().isManual
             Deployment previousDeployment = application.deployments().get(zone);
             Deployment newDeployment = previousDeployment;
             if (previousDeployment == null) {
-                newDeployment = new Deployment(zone, revision, version, clock.instant(), new HashMap<>(), new HashMap<>());
+                newDeployment = new Deployment(zone, revision, version, clock.instant());
```
Comment (bratseth) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java]: Could you encapsulate this stuff into a factory method on Deployment? 

I think all of this (line 334 to 340) should just be
`Deployment newDeployment = application.deployments().getOrDefault(new Deployment(zone)).with(revision).at(time);
`

**Review by bratseth — COMMENTED** [review]
This will be so much better than it was ...
I don't see anything about periodically updating the metrics here. Is the maintenance job already updating these metrics periodically?

**Comment by bratseth** [comment]
> Will commit the maintainer before it is ready to be merged.

Ok, got it. Comments crossed ...
LGTM then!

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java (hunk 1 above)
Comment (smorgrav) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java]: What should we default revision, version and deploy-time to in that scheme? And would that instance be valid (before adding revision, version and time)?

Comment (smorgrav) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java]: Alternatively we could us a builder....

Comment (smorgrav) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java]: I went with a middle ground here - which I think served the intent without considering the validity questions I had.

**Pull request merged**